File: 253-meeting-rooms-ii/253-meeting-rooms-ii.java
class Solution {
    public int minMeetingRooms(int[][] intervals) {
        int n = intervals.length;
        int[] start = new int[n];
        int[] end = new int[n];
        for(int i = 0; i < n; i++) {
            start[i] = intervals[i][0];
            end[i] = intervals[i][1];
        }
        Arrays.sort(start);
        Arrays.sort(end);
        
        int room = 0, ei = 0;
        for(int i = 0; i < n; i++) {
            if(start[i] < end[ei]) {
                room++;
            } else {
                ei++;
            }
        }
        return room;
    }
}

File: 253-meeting-rooms-ii/253-meeting-rooms-ii.py
class Solution:
    def minMeetingRooms(self, intervals: List[List[int]]) -> int:
        time = []
        for interval in intervals:
            time.append((interval[0], 1))
            time.append((interval[1], -1))
        
        time.sort(key = lambda x : (x[0], x[1]))
        
        cnt = 0
        res = 0
        for t in time:
            cnt += t[1]
            res = max(res, cnt)
        return res